WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListItemUiStateHelper.kt

```diff
@@ -233,44 +234,69 @@ class PostListItemUiStateHelper @Inject constructor(
             } else {
-                labels.add(UiStringRes(R.string.post_uploading))
+                labels.add(UiStringRes(string.post_uploading))
             }
-            uploadUiState is UploadingMedia -> labels.add(UiStringRes(R.string.uploading_media))
-            uploadUiState is UploadQueued -> labels.add(UiStringRes(R.string.post_queued))
+            uploadUiState is UploadingMedia -> labels.add(UiStringRes(string.uploading_media))
+            uploadUiState is UploadQueued -> labels.add(UiStringRes(string.post_queued))
             uploadUiState is UploadWaitingForConnection -> {
-                when (uploadUiState.postStatus) {
-                    UNKNOWN, PUBLISHED -> labels.add(UiStringRes(R.string.post_waiting_for_connection_publish))
-                    PRIVATE -> labels.add(UiStringRes(R.string.post_waiting_for_connection_private))
-                    PENDING -> labels.add(UiStringRes(R.string.post_waiting_for_connection_pending))
-                    SCHEDULED -> labels.add(UiStringRes(R.string.post_waiting_for_connection_scheduled))
-                    DRAFT -> labels.add(UiStringRes(R.string.post_waiting_for_connection_draft))
-                    TRASHED -> AppLog.e(
-                            POSTS,
-                            "Developer error: This state shouldn't happen. Trashed post is in " +
-                                    "UploadWaitingForConnection state."
-                    )
+                getWaitingForConnectionStatus(uploadUiState.postStatus)?.let {
+                    labels.add(it)
                 }
             }
-            hasUnhandledConflicts -> labels.add(UiStringRes(R.string.local_post_is_conflicted))
-            hasAutoSave -> labels.add(UiStringRes(R.string.local_post_autosave_revision_available))
+            hasUnhandledConflicts -> labels.add(UiStringRes(string.local_post_is_conflicted))
+            hasAutoSave -> labels.add(UiStringRes(string.local_post_autosave_revision_available))
+        }
+        return labels
+    }
+
+    private fun getWaitingForConnectionStatus(postStatus: PostStatus): UiString? {
+        return when (postStatus) {
+            UNKNOWN, PUBLISHED -> (UiStringRes(string.post_waiting_for_connection_publish))
+            PRIVATE -> (UiStringRes(string.post_waiting_for_connection_private))
+            PENDING -> (UiStringRes(string.post_waiting_for_connection_pending))
+            SCHEDULED -> (UiStringRes(string.post_waiting_for_connection_scheduled))
+            DRAFT -> (UiStringRes(string.post_waiting_for_connection_draft))
+            TRASHED -> {
+                AppLog.e(
+                        POSTS,
+                        "Developer error: This state shouldn't happen. Trashed post is in " +
+                                "UploadWaitingForConnection state."
+                )
+                return null
+            }
         }
+    }
+
+    @Suppress("LongParameterList")
```

Comment:
I know this solution works. I have seen it in usage elsewhere but I know we are not fond of doing suppressions. Part of me feels it would be better if we simply pass the `PostModel` here and create these values within the function to lessen the parameters substantially. Do you have any thoughts on this discourse?

Reply:
I'm not a fan as well, to be honest. I added here because I assumed in this case the long parameter list wasn't making the function more complex or harder to understand (given that the parameters were well named).

But I like your solution 😄  I'll pass the `PostModel` instead of a few parameters.

Reply:
Thanks! I remember one of the concerns for passing around the`PostModel` is that it's quite a big object that contains so much information about the post that we may not require. To ensure we are making the best decision here, I am going to loop in another Android developer who has worked with these kind of decisions. 

Hi @ParaskP7 I remember us making a decision about `@Suppress("LongParameterList")` in one of my PRs a few months back. In this case, we were opting to remove it and pass in a `PostModel` to represent several parameters that were extracted from it. However, we know the `PostModel` is quite a large object. Do you have any thoughts on this? Let me know. Thank you.

Reply:
👋 @jd-alexander @dnalves !

> I remember one of the concerns for passing around thePostModel is that it's quite a big object that contains so much information about the post that we may not require.
> In this case, we were opting to remove it and pass in a PostModel to represent several parameters that were extracted from it. However, we know the PostModel is quite a large object.

Thanks for looping me in to this conversation Joel. I would personally go with passing the whole `PostModel` and thus resolving the `LongParameterList` warning. My reasoning would be:
1. The `PostModel` is already in memory, so I am not sure if it being a large object would degrade performance or cause any trouble. I haven't looked at the code extensively, but are you having concerns about passing this object between screens, thus having is unnecessarily being serialized and deserialised, or is it just being passed from one class/method to another?
2. I am not sure how big could potentially the `PostModel` end up being, thus I might be totally off here, but I would not try to optimize performance preemptively by sending individual properties of this object. However, this is my personal rule of thump, which might very well not apply here, so please take it with a grain of salt.

Let me know if I can further help here or whether I am missing some more context. 🙏

Reply:
👋 @ParaskP7 

Thanks for the quick response on this matter 😄 I agree with your points. About your question:

> [...] are you having concerns about passing this object between screens, thus having is unnecessarily being serialized and deserialised, or is it just being passed from one class/method to another?

This is just from one method to another, so no serialization occurring in this case.

@jd-alexander: I made those changes in https://github.com/wordpress-mobile/WordPress-Android/pull/15351/commits/982bd0fad599a8e2d1e376adf861151d233d9654, already resolving the `LongParameterList` 😄

Reply:
> The PostModel is already in memory, so I am not sure if it being a large object would degrade performance or cause any trouble. I haven't looked at the code extensively, but are you having concerns about passing this object between screens, thus having is unnecessarily being serialized and deserialised, or is it just being passed from one class/method to another?

It's just being passed to another method and yes it is already in memory :) 

> I am not sure how big could potentially the PostModel end up being, thus I might be totally off here, but I would not try to optimize performance preemptively by sending individual properties of this object. However, this is my personal rule of thumb, which might very well not apply here, so please take it with a grain of salt.

> Let me know if I can further help here or whether I am missing some more context. 🙏

The thoughts you shared were spot on. You picked up the context quite quickly, just like an Android API using `requireContext()` 😄 😆 

 Thanks for chiming in @ParaskP7 . I think the current approach here is fine!